sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobApiTests.java

```diff
@@ -1596,6 +1598,58 @@ public void perCallPolicy() {
             client.getPropertiesWithResponse(null, null, null).getHeaders().getValue("x-ms-version"));
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"ENCRYPTIONDATA", "EncryptionData", "eNcRyPtIoNdAtA"})
+    public void encryptionDataCaseInsensitivity(String newKey) {
+        byte[] data = getRandomByteArray(Constants.KB);
+        bec = getEncryptionClient(EncryptionVersion.V2, generateBlobName());
+        bec.upload(BinaryData.fromBytes(data));
+        // change casing of encryption data key
+        Map<String, String> metadata = bec.getProperties().getMetadata();
+        String encryptionData = metadata.get("encryptiondata");
+        Map<String, String> encryptionMetadata = new HashMap<>();
+        encryptionMetadata.put(newKey, encryptionData);
+        bec.setMetadata(encryptionMetadata);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        // call with downloadStream to test code path for non-specified BlobRange
+        bec.downloadStream(os);
+        assertArrayEquals(data, os.toByteArray());
+        // now call downloadStreamWithResponse with BlobRange passed to ensure ranged download is working
+        os = new ByteArrayOutputStream();
+        bec.downloadStreamWithResponse(os, new BlobRange(0, (long) Constants.KB), null, null, false, null,
+            Context.NONE);
+        assertArrayEquals(data, os.toByteArray());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"ENCRYPTIONDATA", "EncryptionData", "eNcRyPtIoNdAtA"})
+    public void encryptionDataCaseInsensitivityAsyncClient(String newKey) {
```

Comment:
Also please make sure to manually run these tests with Fiddler enabled to verify the encryption data metadata key has incorrect case sensitivity and we can still decrypt.